dz_dynamodb/sync_strategies/log_based.py

```diff
@@ -62,30 +65,36 @@ def get_shard_records(streams_client, stream_arn, shard, sequence_number):
         'ShardId': shard['ShardId'],
         'ShardIteratorType': iterator_type
     }
-
+    print(f"Setting max_iteration_for_open_shards to {max_iteration_for_open_shards}")
     if sequence_number:
         params['SequenceNumber'] = sequence_number
 
     shard_iterator = streams_client.get_shard_iterator(**params)['ShardIterator']
-
-    # This will loop indefinitely if called on open shards
     while shard_iterator:
         records = streams_client.get_records(ShardIterator=shard_iterator, Limit=1000)
-
+        # for open shards the loop will not so breaking once iterating limit become 0
+        if max_iteration_for_open_shards==0 and not shard['SequenceNumberRange'].get('EndingSequenceNumber') :
+            break
+        max_iteration_for_open_shards-=1
         for record in records['Records']:
             yield record
-
         shard_iterator = records.get('NextShardIterator')
 
 
-def sync_shard(shard, seq_number_bookmarks, streams_client, stream_arn, projection, deserializer, table_name, stream_version, state):
+def sync_shard(shard, seq_number_bookmarks, streams_client, stream_arn, projection, deserializer, table_name, stream_version, state,max_iteration_for_open_shards):
     seq_number = seq_number_bookmarks.get(shard['ShardId'])
 
     rows_synced = 0
 
-    for record in get_shard_records(streams_client, stream_arn, shard, seq_number):
+    for record in get_shard_records(streams_client, stream_arn, shard, seq_number,max_iteration_for_open_shards):
         if record['eventName'] == 'REMOVE':
             record_message = deserializer.deserialize_item(record['dynamodb']['Keys'])
+            # for open shards their can be error of parsing timestamp
+            if not shard['SequenceNumberRange'].get('EndingSequenceNumber'):
```

Comment:
lets check whats the exact difference in timestamp format in case of open & closed shards & then only merge this.